modules/core/lib/Auth/Process/GenerateAffiliation.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/**
+ * Filter to generate affiliation(s) based on groupmembership attribute
+ *
+ * @author Martin van Es, m7
+ * @package simpleSAMLphp
+ */
+class sspmod_core_Auth_Process_GenerateAffiliation extends SimpleSAML_Auth_ProcessingFilter {
+
+    /**
+    * The attributename we should assign affiliations to (target)
+    */
+    private $attributename = 'eduPersonAffiliation';
+
+    /**
+    * The attributename we should generate affiliations from
+    */
+    private $memberattribute = 'memberOf';
+
+    /**
+    * The required $memberattribute values and target affiliations
+    */
+    private $values = array();
+    
+    /**
+    * Wether $memberattribute should be replaced by target attribute
+    */
+    private $replace = FALSE;
+    
+    /**
+    * Initialize this filter.
+    *
+    * @param array $config  Configuration information about this filter.
+    * @param mixed $reserved  For future use.
+    */
+    public function __construct($config, $reserved) {
+        parent::__construct($config, $reserved);
+
+        assert('is_array($config)');
+
+        /* Validate configuration. */
+        foreach ($config as $name => $value) {
+            if (is_int($name)) {
+                // check if this is an option
+                if ($value === '%replace') {
+                        $this->replace = TRUE;
```

Comment:
[PSR-2](https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md#25-keywords-and-truefalsenull) :wink: